app/ViewStores.tsx

```diff
@@ -1,33 +1,40 @@
 import { ReactElement, useEffect, useState } from "react";
 import { Wrapper } from "./ui/Wrapper";
 import { styled } from "styled-components";
-import { Location } from "./types";
-import { darkGrey, black, greyInactiveButton } from "./ui/colourLibrary";
+import { Store } from "./types";
+import {
+  darkGrey,
+  black,
+  greyInactiveButton,
+  skyBlue,
+} from "./ui/colourLibrary";
 import { LargeButton } from "./ui/LargeButton";
 import { getStores } from "@/server/getStores";
+import { UpdateStore } from "./UpdateStore";
 
-export function ViewStores(props: {
-  onCreateClick: () => void;
-  onStoreClick: () => void;
-}): ReactElement {
-  const { onCreateClick, onStoreClick } = props;
-  const [stores, setStores] = useState<Location[]>([]);
+export function ViewStores(props: { onCreateClick: () => void }): ReactElement {
+  const { onCreateClick } = props;
+  const [stores, setStores] = useState<Store[]>([]);
+  const [storeIsSelected, setStoreIsSelected] = useState(false);
+  const [storeDetails, setPrefilledStoreDetails] = useState<Store>();
   useEffect(() => {
     const fetchData = async () => {
       const result = await getStores();
       setStores(result);
     };
     fetchData();
   }, []);
-
   const displayedStores = stores.map((store) => {
```

Comment:
now that there's a chance that the displayed stores aren't rendered, it's preferable not to calculate these `displayedStores`, ie if there is a selected store, we show the update component which doesn't use `displayedStores`, so we've calculated it for nothing.

Reply:
https://github.com/auberginearms/groceryDatabaseApp/pull/20/commits/7ac3664e395a83e13307075644a97c403a0ed4f3